pool/src/emissions/manager.rs

```diff
@@ -72,20 +72,37 @@ fn do_gulp_emissions(e: &Env, new_emissions: i128) {
     }
     let pool_emissions = storage::get_pool_emissions(e);
     let reserve_list = storage::get_res_list(e);
+    let reserve_flags_summary = storage::get_res_flags_summary(e);
 
     let mut total_share: i128 = 0;
-    for (_res_token_id, res_eps_share) in pool_emissions.iter() {
-        total_share += i128(res_eps_share);
+    for (res_token_id, res_eps_share) in pool_emissions.iter() {
+        let reserve_index = res_token_id / 2;
+        let res_asset_address = reserve_list.get_unchecked(reserve_index);
+
+        if reserve_flags_summary
+            .get(res_asset_address)
+            .unwrap_or(ReserveFlags::Enabled as u32)
+            == ReserveFlags::Enabled as u32
+        {
+            total_share += i128(res_eps_share);
+        }
     }
     for (res_token_id, res_eps_share) in pool_emissions.iter() {
         let reserve_index = res_token_id / 2;
         let res_asset_address = reserve_list.get_unchecked(reserve_index);
-        let new_reserve_emissions = i128(res_eps_share)
-            .fixed_div_floor(total_share, SCALAR_7)
-            .unwrap_optimized()
-            .fixed_mul_floor(new_emissions, SCALAR_7)
-            .unwrap_optimized();
-        update_reserve_emission_eps(e, &res_asset_address, res_token_id, new_reserve_emissions);
+
+        if reserve_flags_summary
+            .get(res_asset_address.clone())
+            .unwrap_or(ReserveFlags::Enabled as u32)
+            == ReserveFlags::Enabled as u32
+        {
+            let new_reserve_emissions = i128(res_eps_share)
+                .fixed_div_floor(total_share, SCALAR_7)
+                .unwrap_optimized()
+                .fixed_mul_floor(new_emissions, SCALAR_7)
+                .unwrap_optimized();
+            update_reserve_emission_eps(e, &res_asset_address, res_token_id, new_reserve_emissions);
```

Comment:
`update_reserve_emissions_eps` loads the reserve config for each asset anyway.

My suggestion:
1. Use `reserve_config` to load the asset's enabled status
2. Pass the `reserve_config` into `update_reserve_emission_eps` instead of the `res_token_id` to avoid loading it twice. `reserve_config.id === res_token_id`

Reply:
This will likely require us to build a `pool_emis_enabled: Vec<(reserve_config, res_eps_share)>` vector during the first pass that determines the `total_share`, to also avoid loading the `reserve_config` twice.